lib/gruf/synchronized_client.rb

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2017-present, BigCommerce Pty. Ltd. All rights reserved
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+require 'concurrent'
+
+module Gruf
+  ##
+  # Ensures that we only have one active call to a given endpoint with a given set of params. This can be useful
+  # to mitigate thundering herds. Cache the result for a period of time so other requests, once unblocked, will get the
+  # same value. These results are expired after the given ttl.
+  #
+  class SynchronizedClient < Gruf::Client
+    ##
+    # Initialize the client and setup the stub
+    #
+    # @param [Module] service The namespace of the client Stub that is desired to load
+    # @param [Hash] options A hash of options for the client
+    # @option options [String] :password The password for basic authentication for the service.
+    # @option options [String] :hostname The hostname of the service. Defaults to linkerd.
```

Comment:
I think you can remove the hostname/password options here from the docblock as their implied through inheritance